bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/BlueZBluetoothDevice.java

```diff
@@ -390,10 +399,54 @@ private BluetoothGattDescriptor getTinybDescriptorByUUID(String uuid) {
         return null;
     }
 
+    private void cancelInactiveCleanupJob() {
+        try {
+            inactiveCleanupJobLock.lock();
+            if (inactiveCleanupJob != null) {
+                inactiveCleanupJob.cancel(true);
+                inactiveCleanupJob = null;
+            }
+        } finally {
+            inactiveCleanupJobLock.unlock();
+        }
+    }
+
+    private void onActivity() {
+        logger.trace("device scanned: {}", device.getAddress());
+        try {
+            inactiveCleanupJobLock.lock();
+            cancelInactiveCleanupJob();
+            inactiveCleanupJob = scheduler.schedule(() -> {
+                synchronized (BlueZBluetoothDevice.this) {
+                    if (device != null) {
+                        logger.debug("Removing device '{}' due to inactivity", device.getAddress());
```

Comment:
Isn't the TTL on discovery-results enough? I remember from the network binding that it will just remove them after some time automatically.

Reply:
At least that was the whole idea of the TTL...

Reply:
well yes, the TTL will handle the inbox side of things, but it won't handle the never ending discovery window we are using in the bluez binding. With just the TTL the use would be none the wiser to the endlessly growing discovered mac address list being tracked by bluez. I added the inactivityCleanupJob in order to combat devices with periodic changing mac addresses and or devices that are no longer reachable.

Reply:
Ok, so the main issue is that bluez caches all discovered devices (and does not have an internal TTL) indefinitely, even if they have not been seen for a while - and as we request the current list from bluez, we see the same old devices turning up again. Did I get this right?